lib/IMAP/Sync/Synchronizer.php

```diff
@@ -97,8 +97,8 @@ public function sync(Horde_Imap_Client_Base $imapClient,
 			throw $e;
 		}
 
-		$newMessages = $this->messageMapper->findByIds($imapClient, $request->getMailbox(), new Horde_Imap_Client_Ids($newUids), $userId);
-		$changedMessages = $this->messageMapper->findByIds($imapClient, $request->getMailbox(), new Horde_Imap_Client_Ids($changedUids), $userId);
+		$newMessages = $this->messageMapper->findByIds($imapClient, $request->getMailbox(), $newUids, $userId);
+		$changedMessages = $this->messageMapper->findByIds($imapClient, $request->getMailbox(), $changedUids, $userId);
```

Comment:
@st3iny this is the problematic call. You can mod `getChangedMessageUids` to just return ``$request->getUids()``. It will fake that all messages changed and therefore trigger a re-fetch.